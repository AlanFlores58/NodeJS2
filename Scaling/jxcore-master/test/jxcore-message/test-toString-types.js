// Copyright & License details are available under JXCORE_LICENSE file


var print = function(obj, name) {
  var str = "";
  if (obj === null) str = "[null]"; else
  if (obj === undefined) str = "[undefined]"; else
  str = obj.toString() || "[empty string]";
  console.log("\t`" + name + "` toString() =", str);
};

console.log("\nnamespace:", "types");
print("str", "some string");
print("", "empty string");

print(0, "number 0");
print(1, "number 1");
print(2.7, "number 2.7");

print(true, "boolean true");
print(false, "boolean false");

print([], "empty array");
print([ 1, "two"], "non empty array");

print({}, "empty object");
print({ x : 1 }, "non empty object");

print(null, "null");
print(undefined, "undefined");

print(function() {}, "function");
