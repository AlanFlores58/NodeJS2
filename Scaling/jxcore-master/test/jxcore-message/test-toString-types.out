
namespace: types
	`some string` toString() = str
	`empty string` toString() = [empty string]
	`number 0` toString() = 0
	`number 1` toString() = 1
	`number 2.7` toString() = 2.7
	`boolean true` toString() = true
	`boolean false` toString() = false
	`empty array` toString() = [empty string]
	`non empty array` toString() = 1,two
	`empty object` toString() = [object Object]
	`non empty object` toString() = [object Object]
	`null` toString() = [null]
	`undefined` toString() = [undefined]
	`function` toString() = function () {}
